surveys/micro/2025-06-compiler-performance.md

```diff
@@ -0,0 +1,274 @@
+# Survey questions
+
+The goal of this survey is to figure out how much do Rust users struggle with compilation times and which compilation workflows are the most relevant for them. We would like to use this data so that we can focus the efforts of Rust compiler contributors on the areas of compilation performance that matter the most to our users.
+
+Your feedback is much appreciated. This survey is fully anonymous. The Rust survey team will go through the answers and release a summary on the Rust blog after the survey is complete.
+
+## User classification
+
+### Do you use Rust?
+
+Type: select one
+
+- Yes, I use Rust [`NEXT`](#do-you-code-in-rust-at-a-company)
+- No, I don't currently use Rust, but I have in the past [`NEXT`](#were-long-compilation-times-the-primary-reason-why-you-stopped-using-rust)
+- No, I have never used Rust [`NEXT`](#other)
+
+### Were long compilation times the primary reason why you stopped using Rust?
+
+Type: select one
+
+- Yes
+- No, but it was one of the reasons why I stopped using Rust
+- No
+
+### Which operating systems do you use regularly for Rust development?
+
+**Note**: this is specifically about which systems you personally use for development, *not* all the
+systems you target.
+
+Type: select all that apply (optional)
+
+- Linux
+- Windows
+- Windows Subsystem for Linux
+- macOS
+- Other (open response)
+
+### Do you code in Rust at a company?
+
+If yes, what is the approximate size of the company?
+
+Type: select one (optional)
+
+- No, I do not code in Rust in a company
+- Yes (1-50 employees)
+- Yes (51-500 employees)
+- Yes (501-5000 employees)
+- Yes (more than 5000 employees)
+
+## Workflows
+
+### Which builds system do you use to build Rust code?
+
+Type: select all that apply (optional)
+
+- I use Cargo
+- I use some other build system
+- I combine Cargo and another build system
+- If you use other build system(s), which ones do you use? (open response)
+
+### How do you build your Rust projects?
+
+Type: select all that apply (optional)
+
+- On my local computer
+- On CI infrastructure
+- On a remote or cloud server
+- Other (open response)
+
+### Which compilation workflows are the most important to you?
+
+Please rate how important are the following compilation workflows to you, and how much do you currently struggle with their performance currently.
+
+Type: matrix (optional)
+
+Workflows:
+
+- Unoptimized rebuild (change code, rebuild without optimizations)
+- Optimized rebuild (change code, rebuild with optimizations)
+- Workspace rebuild (change a crate which causes multiple other crates in your workspace to be rebuilt)
+- Clean unoptimized build (build a crate graph from scratch)
+- Clean optimized build (build a crate graph from scratch)
+- Build in a CI workflow
+
+Priority:
+
+- Not so important for me
+- Important, compile times are acceptable for me
+- Important, compile times are a blocker for me
```

Comment:
"compile times are a blocker" is too extreme. I suspect anyone actually blocked by compile times won't be answering this question and most people will say they aren't "acceptable" but not so bad as to block.